src/main/java/com/github/mmolimar/kafka/connect/fs/FsSourceTask.java

```diff
@@ -78,13 +78,18 @@ public void start(Map<String, String> properties) {
             List<SourceRecord> totalRecords = filesToProcess().map(metadata -> {
                 List<SourceRecord> records = new ArrayList<>();
                 try (FileReader reader = policy.offer(metadata, context.offsetStorageReader())) {
+                    if(reader == null){
+                        log.info("Skipping processing file {} as it is unchanged", metadata);
+                        return records;
+                    }
                     log.info("Processing records for file {}.", metadata);
                     while (reader.hasNext()) {
                         records.add(convert(metadata, reader.currentOffset() + 1, reader.next()));
                     }
                 } catch (ConnectException | IOException e) {
                     //when an exception happens reading a file, the connector continues
                     log.error("Error reading file [{}]. Keep going...", metadata.getPath(), e);
+                    return new ArrayList<SourceRecord>();
```

Comment:
Why returning here an empty list?

Reply:
See the comment here: https://github.com/mmolimar/kafka-connect-fs/pull/60#discussion_r429832760

If we commit offsets for a file containing the `fileSizeBytes` but we only processed part way through the file due to an error we will skip the data from the rest of the file. Instead we should only commit the offsets for that file if the file was processed entirely.